File: reservations - SPA/reservations-spa/src/app/_models/Office.ts
import { Address } from './Address';
import { Desk } from './desk';
import { Room } from './room';

export class Office{
    public id:string;
    public userId:string;
    public name: string;
    public address:Address;
    public rooms:Room[];
    public email:string;
    public desks:Desk[];
    public description:string;
    public phoneNumber:string;

    constructor(name:string,address:Address,rooms:Room[],desks:Desk[],description:string, email:string, phoneNumber:string){
        this.name = name;
        this.address = address;
        this.rooms = rooms;
        this.desks = desks;
        this.description = description;
        this.email = email;
        this.phoneNumber = phoneNumber;
    }
}

File: reservations - API/reservations/Reservations.Core/Domain/Office.cs
﻿using ProjectCalculator.Core.Domain;
using System;
using System.Collections.Generic;
using System.Text;

namespace Reservations.Core.Domain
{
    public class Office
    {
        public Guid Id { get; protected set; }
        public Guid UserId { get; protected set; }
        public Address Address { get; protected set; }
        public string Name { get; protected set; }
        public string Description { get; set; }
        public List<Room> Rooms { get; protected set; }
        public List<Desk> Desks { get; protected set; }

        protected Office() { }

        public Office(Guid officeId,Guid userId, Address address, string name, string description)
        {
            Id = officeId;
            UserId = userId;
            Address = address;
            Name = name;
            Description = description;
        }
    }
}
